api/api.go

```diff
@@ -292,6 +295,27 @@ func proxyContainer(c *context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Proxy a request to the right node
+func proxyImage(c *context, w http.ResponseWriter, r *http.Request) {
+	name := mux.Vars(r)["name"]
+
+	for _, node := range c.cluster.Nodes() {
+		for _, image := range node.Images() {
```

Comment:
In the same way we have `cluster.Containers()` and `cluster.Container(name)`, how about `node.Images()` and `node.Image(name)`?

In that case you could move the image matching code back there.